src/core/fbcontainer.h

```diff
@@ -155,6 +167,46 @@ namespace forte {
         TFBContainerList mChildren; //!< List of children (i.e, fbs or subapplications in this container)
     };
 
+    template<typename T>
+    class CInternalFB {
+        static_assert(std::is_base_of_v<CFunctionBlock, T>, "T must be a CFunctionBlock");
+      public:
+        CInternalFB(CStringDictionary::TStringId paInstanceNameId, CFBContainer &paContainer) :
+                mFB(std::make_unique<T>(paInstanceNameId, paContainer)) {
+          paContainer.addFB(*mFB);
+        }
+
+        CInternalFB(CStringDictionary::TStringId paInstanceNameId, const char *paConfigString,
+                    CFBContainer &paContainer) :
+                mFB(std::make_unique<T>(paInstanceNameId, paContainer)) {
+          mFB->configureFB(paConfigString);
+          paContainer.addFB(*mFB);
+        }
+
+        ~CInternalFB() {
+          mFB->getParent().removeFB(*mFB);
+        }
+
+        T *get() noexcept {
```

Comment:
do we really need access to the pointer?

Reply:
When we do need a direct pointer to the wrapped FB class, this avoids constructs such as `&(*fb_FB1)`.

Reply:
make sense